allennlp/models/multitask.py

```diff
@@ -0,0 +1,196 @@
+        inputs in the same batch. If we see overlapping keys, we will crash.  If you want to be able
+        to do this kind of mixed training in the same batch, you need to handle that in your data
+        code, not here; we won't handle complex batching inside this model.
+    allowed_arguments: `Dict[str, Set[str]]`, optional (default = `inferred`)
+        The list of arguments that should be passed from `**kwargs` to the `forward` method for the
+        backbone and each head.  If you provide this, the keys in here should match the keys given
+        in the `heads` parameter, plus a "backbone" key for the backbone arguments.  If not given,
+        we will use the `inspect` module to figure this out.  The only time that this inference
+        might fail is if you have optional arguments that you want to be ignored, or
+        something.  You very likely don't need to worry about this argument.
+    initializer: `InitializerApplicator`, optional (default=`InitializerApplicator()`)
+        If provided, will be used to initialize the model parameters.
+    """
+
+    def __init__(
+        self,
+        vocab: Vocabulary,
+        backbone: Backbone,
+        heads: Dict[str, Head],
+        *,
+        loss_weights: Dict[str, float] = None,
+        arg_name_mapping: Dict[str, Dict[str, str]] = None,
+        allowed_arguments: Dict[str, Set[str]] = None,
+        initializer: InitializerApplicator = InitializerApplicator(),
+        **kwargs,
+    ):
+        super().__init__(vocab, **kwargs)
+        self._backbone = backbone
+        self._heads = torch.nn.ModuleDict(heads)
+        self._arg_name_mapping = arg_name_mapping or defaultdict(dict)
+
+        self._allowed_arguments = allowed_arguments or {
+            "backbone": get_forward_arguments(backbone),
+            **{key: get_forward_arguments(heads[key]) for key in heads},
+        }
+        self._loss_weights = loss_weights or defaultdict(lambda: 1.0)
+        self._active_heads: List[str] = None
+        initializer(self)
+
+    def set_active_heads(self, active_heads: List[str]) -> None:
+        """
+        By default, the `MultiTaskModel` will try to infer which heads to run from the arguments
+        passed to `forward`.  During training, we will only run a head if we have all of its
+        arguments, including optional arguments, which typically means the argument is the
+        prediction target; if we don't have it, we typically can't compute a loss, so running during
+        training is pointless.  During evaluation, we will run all heads.
+
+        If you want to limit which heads are run during evaluation, or if the inference for which
+        task to run during training is incorrect (e.g., if your head has multiple optional
+        arguments, and only some are actually required to compute a loss), then you can use this
+        method to override our inference and force the use of whatever set of hads you want.
```

Comment:
```suggestion
        method to override our inference and force the use of whatever set of heads you want.
```

Reply:
My breaking `e` key strikes again.  I need to get my keyboard fixed...